addons/knative/enable

```diff
@@ -4,30 +4,111 @@ set -e
 
 source $SNAP/actions/common/utils.sh
 
-echo "Enabling Knative"
-
-"$SNAP/microk8s-enable.wrapper" istio
-
+KNATIVE_VERSION="1.6.0"
+KNATIVE_FUNC_VERSION="0.25.0"
+KNATIVE_SERVING=true
+KNATIVE_EVENTING=false
 KUBECTL="$SNAP/kubectl --kubeconfig=${SNAP_DATA}/credentials/client.config"
 
-declare -a yamls=("https://github.com/knative/serving/releases/download/v0.24.0/serving-crds.yaml"
-                  "https://github.com/knative/eventing/releases/download/v0.24.0/eventing-crds.yaml"
-                  "https://github.com/knative/serving/releases/download/v0.24.0/serving-core.yaml"
-                  "https://github.com/knative/net-istio/releases/download/v0.24.0/net-istio.yaml"
-                  "https://github.com/knative/serving/releases/download/v0.24.0/serving-default-domain.yaml"
-                  "https://github.com/knative/eventing/releases/download/v0.24.0/eventing-core.yaml"
-                  "https://github.com/knative/eventing/releases/download/v0.24.0/in-memory-channel.yaml"
-                  "https://github.com/knative/eventing/releases/download/v0.24.0/mt-channel-broker.yaml"
-                 )
-
-for yaml in "${yamls[@]}"
+for i in "$@"
 do
-   $KUBECTL apply -f "$yaml"
-   sleep 3
+case $i in
+    --no-serving)
+    KNATIVE_SERVING=false
+    shift # past argument
+    ;;
+    --eventing)
+    KNATIVE_EVENTING=true
+    shift # past argument
+    ;;
+    *)
+          # unknown option
+    ;;
+esac
 done
 
+#requirement
+"$SNAP/microk8s-enable.wrapper" dns
+
+echo "Enabling Knative ${KNATIVE_VERSION}"
+$KUBECTL apply -f https://github.com/knative/operator/releases/download/knative-v$KNATIVE_VERSION/operator.yaml
+
+if [ "$KNATIVE_SERVING" = "true" ]; then
+    echo "Installing Knative Serving ${KNATIVE_VERSION}"
+$KUBECTL apply -f- << EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: knative-serving
+---
+apiVersion: operator.knative.dev/v1beta1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+  namespace: knative-serving
+spec:
+  ingress:
+    kourier:
+      enabled: true
+  config:
+    network:
+      ingress-class: "kourier.ingress.networking.knative.dev"
+EOF
+fi
+
+if [ "$KNATIVE_EVENTING" = "true" ]; then
+   echo "Installing Knative Eventing ${KNATIVE_VERSION}"
+$KUBECTL apply -f- << EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: knative-eventing
+---
+apiVersion: operator.knative.dev/v1beta1
+kind: KnativeEventing
+metadata:
+  name: knative-eventing
+  namespace: knative-eventing
+EOF
+fi
+
+# Handle Knative CLIs
+if [ ! -f "${SNAP_DATA}/bin/kn" ]; then
```

Comment:
```suggestion
if [ ! -f "${SNAP_COMMON}/plugins/kn" ]; then
```
Install binaries in `${SNAP_COMMON}/plugins`, and they can then be called from the main microk8s entrypoint:

```
$ microk8s
....
Available subcommands from addons are:
	bin
	kn
	kn-admin
	kn-event
```

Reply:
Thanks I was looking for this documentation, I asked here https://github.com/canonical/microk8s-community-addons/issues/65

Copied from cilium and linked:
https://github.com/canonical/microk8s-community-addons/blob/main/addons/cilium/enable#L108
https://github.com/canonical/microk8s-community-addons/blob/main/addons/linkerd/enable#L20